CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using CoffeeMachineSimulator.Services.Interfaces;
+using CoffeeMachineSimulator.Services.Models;
+using FizzWare.NBuilder;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CoffeeMachineSimulator.Services.Services
+{
+    public class CoffeeService : ICoffeeService
+    {
+        private readonly List<CoffeeModel> coffeeModels;
+
+        public CoffeeService()
+        {
+            coffeeModels = Builder<CoffeeModel>.CreateListOfSize(10).Build().ToList();
+        }
+
+        public void AddCoffee(CoffeeModel coffeeToAdd)
+        {
+            if (coffeeToAdd == null) { throw new Exception("You should not add null entries!"); }
+            foreach(CoffeeModel coffee in coffeeModels)
+            { if (coffee.Id == coffeeToAdd.Id) { throw new Exception("Already exists an entry with the same Id"); } }
+            if (!(coffeeToAdd.Id == Guid.Empty || coffeeToAdd.Name == null || coffeeToAdd.Price == 0.0f))
```

Comment:
For Long data types (or any data types) use for comparison their .MaxValue or .MinValue properties. ex: coffeeToAdd.Price == long.MinValue;